module/Motion/DistanceStraight.cpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file   DistanceStraight.cpp
+ * @brief  目標距離まで直進するクラス
+ * @author bizyutyu
+ */
+
+#include "DistanceStraight.h"
+using namespace std;
+
+DistanceStraight::DistanceStraight(double _targetDistance, double _targetSpeed)
+  : Straight(_targetSpeed), targetDistance(_targetDistance)
+{
+}
+
+bool DistanceStraight::isMetPreCondition()
+{
+  char buf[LARGE_BUF_SIZE];
+  // 目標速度値が0の場合は終了する
+  if(targetSpeed == 0) {
+    logger.logWarning("The targetSpeed value passed to DistanceStraight is 0");
+    return false;
+  }
+  // 目標距離の値が0以下の場合はwarningを出して終了する
+  if(targetDistance <= 0) {
+    snprintf(buf, LARGE_BUF_SIZE, "The targetDistance value passed to DistanceStraight is %.2f",
+             targetDistance);
+    logger.logWarning(buf);
+    return false;
+  }
+
+  return true;
+}
+
+bool DistanceStraight::isMetContinuationCondition()
+{
+  // 現在の走行距離を取得する
+  currentRightMotorCount = measurer.getRightCount();
+  currentLeftMotorCount = measurer.getLeftCount();
+  currentRightDistance = Mileage::calculateWheelMileage(currentRightMotorCount);
+  currentLeftDistance = Mileage::calculateWheelMileage(currentLeftMotorCount);
```

Comment:
currentRightDistance，currentLeftDistanceはそれぞれ `Mileage::calculateWheelMileage(currentRightMotorCount)`
`Mileage::calculateWheelMileage(currentLeftMotorCount)`
で値をセットしているだけなので，インスタンス変数として持たせる必要がないかもしれません．
見落としていたらすみません．
```suggestion
  currentRightMotorCount = measurer.getRightCount();
  currentLeftMotorCount = measurer.getLeftCount();
```